File: app/src/main/java/com/example/administrator/learnandbase/interfaces/UiOperation.java
package com.example.administrator.learnandbase.interfaces;

import android.view.View;
import android.view.View.OnClickListener;

public interface UiOperation extends OnClickListener {

    int getLayoutResId();

    void initData();

    void initView();

    void initListener();

    void onClick(View v, int id);
}

File: app/src/main/java/com/example/administrator/learnandbase/interfaces/UiOperation.kt
package com.example.administrator.learnandbase.interfaces

import android.view.View
import android.view.View.OnClickListener

interface UiOperation : OnClickListener {

    val layoutResId: Int

    fun initOnCreate() {}

    fun initOnClick(view: View, id: Int = view.id) {}
}